core/common/src/internal/format/parser/ParserOperation.kt

```diff
@@ -139,6 +139,118 @@ internal class UnconditionalModification<Output>(
     }
 }
 
+internal class TimeZoneParserOperation<Output>(
+    private val setter: AssignableField<Output, String>
+) : ParserOperation<Output> {
+
+    override fun consume(storage: Output, input: CharSequence, startIndex: Int): ParseResult {
+        if (startIndex >= input.length) return ParseResult.Error(startIndex) { "Unexpected end of input" }
```

Comment:
I think that this line should either mention that a timezone is expected, or be removed altogether (as the error case of empty input is covered later anyway).